### PR TITLE
refactor: replace isValid from the datePicker

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes: #2347
Part of #1844 

Now that we have a validator in our codebase, we can replace the moment's isValid in the datepicker.

_Video of the datepicker uploading soon_

### CODE REVIEW

File: src/shared/components/dateRangePicker/DatePicker.tsx [hunk 1]
```diff
@@ -38,25 +39,25 @@ interface State {
...
-    moment(d, 'YYYY-MM-DD HH:mm', true).isValid() ||
-    moment(d, 'YYYY-MM-DD HH:mm:ss', true).isValid() ||
-    moment(d, 'YYYY-MM-DD HH:mm:ss.SSS', true).isValid() ||
-    moment(d, 'YYYY-MM-DD', true).isValid() ||
-    moment(d).toISOString() === d
+    isValid(d, 'YYYY-MM-DD HH:mm') ||
+    isValid(d, 'YYYY-MM-DD HH:mm:ss') ||
+    isValid(d, 'YYYY-MM-DD HH:mm:ss.SSS') ||
+    isValid(d, 'YYYY-MM-DD') ||
+    new Date(d).toISOString() === d
```
Comment (ChitlangeSahas) [inline src/shared/components/dateRangePicker/DatePicker.tsx:46]: Technically wasn't an `isValid` function change, but slipped this in.

**Review by TCL735 — APPROVED** [review]

**Review by hoorayimhelping — COMMENTED** [review]
a comment about, you guessed it, adding tests 😆

File: src/shared/components/dateRangePicker/DatePicker.tsx (hunk 1 above)
Comment (ChitlangeSahas) [reply src/shared/components/dateRangePicker/DatePicker.tsx]: Had to put it back, there is some stuff with datepicker that would need its own issue and PR.

File: src/shared/components/dateRangePicker/DatePicker.tsx [hunk 2]
```diff
@@ -38,25 +39,25 @@ interface State {
...
 const isValidRTC3339 = (d: string): boolean => {
   return (
-    moment(d, 'YYYY-MM-DD HH:mm', true).isValid() ||
-    moment(d, 'YYYY-MM-DD HH:mm:ss', true).isValid() ||
-    moment(d, 'YYYY-MM-DD HH:mm:ss.SSS', true).isValid() ||
-    moment(d, 'YYYY-MM-DD', true).isValid() ||
+    isValid(d, 'YYYY-MM-DD HH:mm') ||
+    isValid(d, 'YYYY-MM-DD HH:mm:ss') ||
+    isValid(d, 'YYYY-MM-DD HH:mm:ss.sss') ||
+    isValid(d, 'YYYY-MM-DD') ||
```
[the remark this answers is not in the document]
Comment (ChitlangeSahas) [reply src/shared/components/dateRangePicker/DatePicker.tsx:45]: I could add this to `src/shared/utils/dateTimeUtils` file!

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 58 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2348)

**Review by hoorayimhelping — APPROVED** [review]
Sahas and I chatted - this is only used by the DatePicker (and it's also not a valid RFC3339 date, and it's also called RTC3339 (😆)) and it's backed up by e2e tests, we don't have to block on writing tests for this functionality, but we should add them soon

**Pull request merged**